es_ES/faq.md

```diff
@@ -0,0 +1,625 @@
+[Iterable](https://docs.flutter.io/flutter/dart-core/Iterable-class.html) de Dart,
+que hace un uso intensivo del estilo funcional (map, reduce, where, etc.), es
+frecuentemente utilizado para procesar listas de valores en el framework.
+
+* Programación event-driven: las interacciones del usuario están representadas por objetos de eventos que se envían a callbacks registradas con controladores de eventos. Actualizaciones de pantalla son activados por un mecanismo de callback similar. La clase [Listenable](https://docs.flutter.io/flutter/foundation/Listenable-class.html), que se utiliza como base del sistema de animación, formatea un modelo de suscripción para eventos con múltiples oyentes.
+
+* Programación orientada a objetos basada en clases: la mayoría de las API del Framework se construyen usando clases con herencia. Usamos un enfoque por el cual definimos API de muy alto nivel en nuestras clases base, luego especialícelas iterativamente en subclases. Por ejemplo, nuestros objetos de renderizado tienen una clase base ([RenderObject](https://docs.flutter.io/flutter/rendering/RenderObject-class.html)) eso es agnóstico con respecto al sistema de coordenadas, y luego tenemos una subclase ([RenderBox](https://docs.flutter.io/flutter/rendering/RenderBox-class.html)) que introduce la opinión de que la geometría debe basarse en el cartesiano sistema de coordenadas (x/ancho y/ alto).
+
+* Programación orientada a objetos basada en prototipos: la clase [ScrollPhysics](https://docs.flutter.io/flutter/widgets/ScrollPhysics-class.html) se instancia para componer la física que se aplica al desplazamiento dinámicamente en tiempo de ejecución. Esto permite que el sistema componga, por ejemplo, paginación física con física específica de plataforma, sin que la plataforma tenga que ser seleccionado en tiempo de compilación.
+
+* Programación imperativa: programación imperativa directa, por lo general emparejado con estado encapsulado dentro de un objeto, se usa donde proporciona la solución más intuitiva. Por ejemplo, las pruebas están escritas en un estilo imperativo, primero describiendo la situación bajo prueba, luego enumerando las invariantes que la prueba debe coincidir, luego avanzar el reloj o insertar eventos según sea necesario para la prueba.
+
+* Programación reactiva: el widget y los árboles de elementos a veces se describen como reactivo, porque las nuevas entradas proporcionadas en el constructor de un widget son inmediatamente propagado como cambios a widgets de nivel inferior por el método de compilación del widget, y los cambios realizados en los widgets inferiores (p. ej., en respuesta a la entrada del usuario) se propagan copia de seguridad del árbol a través de controladores de eventos. Aspectos de reactivos funcionales y
+imperativo-reactivo están presentes en el marco, en función de las necesidades de la
+widgets. Widgets con métodos de compilación que consisten en solo una expresión
```

Comment:
Widgets with build methods that consist of just an expression describing how the widget reacts to changes in its configuration are functional reactive widgets (e.g., the Material Divider class). Widgets whose build methods construct a list of children over several statements, describing how the widget reacts to changes in its configuration, are imperative reactive widgets (e.g., the Chip class).

Widgets con un método build que consiste en solo una expresión describiendo como el widget reacciona a cambios en su configuración son widgets funcionales reactivos (ej, la clase Material Divider). Widgets cuyos métodos build construyen una lista de hijos sobre varias declaraciones, describiendo como el widget reacciona a los cambios en su configuración , son widgets imperativos reactivos (ej la clase Chip).